src/components/structures/GroupView.js

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 Vector Creations Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import MatrixClientPeg from '../../MatrixClientPeg';
+import sdk from '../../index';
+import { sanitizedHtmlNode } from '../../HtmlUtils';
+import { _t } from '../../languageHandler';
+
+
+module.exports = React.createClass({
+    displayName: 'GroupView',
+
+    propTypes: {
+        groupId: React.PropTypes.string.isRequired,
+    },
+
+    getInitialState: function() {
+        return {
+            phase: "GroupView.LOADING",  // LOADING / DISPLAY / ERROR / NOT_FOUND
+            summary: null,
+            error: null,
+        };
+    },
+
+    componentWillMount: function() {
+        this.setState({
+            phase: "GroupView.LOADING",
+            summary: null,
+        })
+        this._loadGroupFromServer(this.props.groupId)
+    },
+
+    componentWillReceiveProps: function(new_props) {
+        if (this.props.groupId != new_props.groupId) {
+            this.setState({
+                phase: "GroupView.LOADING",
+                summary: null,
+            })
+            this._loadGroupFromServer(new_props.groupId);
+        }
+    },
+
+    _loadGroupFromServer: function(groupId) {
+        MatrixClientPeg.get().getGroupSummary(groupId).done((res) => {
+            this.setState({
+                phase: "GroupView.DISPLAY",
+                summary: res,
+            });
+        }, (err) => {
+            this.setState({
+                phase: err.httpStatus == 404 ? "GroupView.NOT_FOUND" :"GroupView.ERROR",
```

Comment:
It seems bad to derive the phase from the error _and_ pass the error through in the state. This leaves the potential for the two to get out of sync